apps/backend/src/usecases/owner/get-pet/get-pet.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { MongoPet, PetDocument } from '../../../infra/mongo/schemas/pet.schema';
+
+type Output = {
+  name: string;
+  breed: string;
+  age: string;
+  type: string;
+  gender: string;
+  size: string;
+  observations: string;
+  photos: string[];
+};
+
+@Injectable()
+export default class GetPet {
+  constructor(
+    @InjectModel(MongoPet.name)
+    private readonly petModel: Model<PetDocument>,
```

Comment:
Instead of passing the Model directly to the useCase, you could create a repository that uses Mongoose to access the data and pass that repository to the useCase, ensuring that the application's useCase isn't dependent on Mongoose. This approach also allows you to swap in another repository that uses a different ORM to access the data in the future if necessary. Does that make sense?

Reply:
thanks for commenting! the "model" here is a database model, not an entity model. the repository pattern doesn't play well here because this use case is a reading model, it just grabs data and returns it to whoever needs it. the repository pattern is directly related to how aggregates are persisted, so here we're not mutating nor persisting anything.

with regards to uncoupling it from mongoose, I totally agree and have done so for the `find-nearest-pets`. since `get-pet` is just being used for testing, I skipped this for now.